File: 151-reverse-words-in-a-string/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        List <String> tokens = new ArrayList<> (); 
        String [] words = s.trim().split("\\s+");
        StringBuilder result = new StringBuilder();
        for (int i = words.length-1; i >= 0 ; i--){
            result.append(words[i]);
            if (i != 0) result.append(" ");
        }
        return result.toString();
    }
}

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        stringstream ss(s);
        string word;
        vector<string> words;
        while(ss >> word){
            words.push_back(word);
        }
        string ans;
        for(int i = words.size() - 1; i >= 0; i--){
            if(!ans.empty()){
                ans += " ";
            }
            ans += words[i];
        }
        return ans;
    }
};